src/avatar-service.cpp

```diff
@@ -229,12 +302,14 @@ QImage AvatarService::getAvatar(const QString& email)
 {
     QImage img = loadAvatarFromLocal(email);
 
-    if (img.isNull()) {
+    // update all avatars if feature autoupdate enabled or img is null
+    if (autoupdate_db_ || img.isNull()) {
```

Comment:
Each time `AvatarService::getAvatar` is called (it's called frequently when painting activities tab), a new GetAvatarRequest would be sent (if there isn't one in the queue), this is not good. We should only send a `GetAvatarRequest` either:

- We have no avatar for the current account yet, or
- We have not checked avatars for a period of time, for example 5 minutes.